### PR TITLE
Updates to better support Rosetta2 runtimes, and MSL 2.4 and 2.3 versions.

### PR DESCRIPTION
**Pull request by billhollings**
- Do not use `MTLEvent` for `VkSemaphore` under Rosetta2.
- Remove compile test for `MVK_MACOS_APPLE_SILICON` and `MVK_APPLE_SILICON` when testing for Apple GPU families, to allow x86 builds to test for Apple GPU under Rosetta2.
- Simplify identifying M1 GPU. All M1 SoCs currently support the A14 (Apple7) GPU.
- Support compiling MSL 2.4 in runtime pipelines and `MoltenVKShaderConverterTool`.
- Fix issue where MSL 2.3 only available on Apple Silicon, even on macOS.
- Update to latest SPIRV-Cross (unrelated to Rosetta2).

Fixes issue #1482.

### CODE REVIEW

**Review by cdavis5e — APPROVED** [review]

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm [hunk 1]
```diff
@@ -2480,7 +2494,7 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 	if (supportsMTLGPUFamily(Apple3)) { mtlFam = MTLGPUFamilyApple3; }
 	if (supportsMTLGPUFamily(Apple4)) { mtlFam = MTLGPUFamilyApple4; }
 	if (supportsMTLGPUFamily(Apple5)) { mtlFam = MTLGPUFamilyApple5; }
-#if MVK_IOS || MVK_MACOS_APPLE_SILICON
+#if MVK_XCODE_12
 	if (supportsMTLGPUFamily(Apple6)) { mtlFam = MTLGPUFamilyApple6; }
 	if (supportsMTLGPUFamily(Apple7)) { mtlFam = MTLGPUFamilyApple7; }
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:2499]: This is part of what's breaking tvOS.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm [hunk 2]
```diff
@@ -2808,7 +2822,7 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 	logMsg += "\n\tsupports the following Metal Versions, GPU's and Feature Sets:";
 	logMsg += "\n\t\tMetal Shading Language %s";
 
-#if MVK_IOS || MVK_MACOS_APPLE_SILICON
+#if MVK_XCODE_12
 	if (supportsMTLGPUFamily(Apple7)) { logMsg += "\n\t\tGPU Family Apple 7"; }
 	if (supportsMTLGPUFamily(Apple6)) { logMsg += "\n\t\tGPU Family Apple 6"; }
 #endif
```
Comment (cdavis5e) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:2828]: This is the other part.

**Review by cdavis5e — CHANGES_REQUESTED** [review]
Sorry, I take that back.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm (hunk 1 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:2499]: Thanks. Yeah. Working on the various combinations of platforms and Xcode versions. We can now add Apple8 on iOS too I see.

Will submit an update soon.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm (hunk 2 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: Fixed.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm (hunk 1 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm]: Fixed.

**Review by cdavis5e — APPROVED** [review]

**Pull request merged**